src/App.tsx

```diff
@@ -47,49 +62,110 @@ export function getReorderedGoods(
 //   sortType: SortType,
 // };
 
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends Component<{}, ReorderOptions> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
+
+  handleSortType = (order: SortType) => {
+    this.setState({ sortType: order });
+  };
+
+  handleOrderChange = () => {
+    this.setState((state) => ({
+      ...state,
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  render(): ReactNode {
+    const {
+      state,
+      handleSortType,
+      handleOrderChange,
+      handleReset,
+    } = this;
+
+    const {
+      sortType,
+      isReversed,
+    } = state;
+
+    const {
+      ALPHABET,
+      LENGTH,
+    } = SortType;
+
+    const goods = getReorderedGoods(goodsFromServer, state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-info',
+              { 'is-light': sortType !== ALPHABET },
+            )}
+            onClick={() => handleSortType(ALPHABET)}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-success',
+              { 'is-light': sortType !== LENGTH },
+            )}
+            onClick={() => handleSortType(LENGTH)}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button',
+              'is-warning ',
+              { 'is-light': !isReversed },
+            )}
+            onClick={handleOrderChange}
+          >
+            Reverse
+          </button>
+
+          {
+            (sortType !== SortType.NONE || isReversed) && (
```

Comment:
will be better to create separate boolean variable before the `return` statement with the result of this check instead of combining several checks directly in `jsx`
in this way you code will be clearer